components/github/github.app.js

```diff
@@ -32,15 +54,126 @@ module.exports = {
       opts.url = `https://api.github.com${path[0] === "/" ? "" : "/"}${path}`
       return await axios(opts)
     },
-    async getRepos({ page }) {
+    async generateSecret() {
+      return "" + Math.random();
+    },
+    async getBranches(opts) {
+      const {
+        page,
+        repoFullName,
+      } = opts
       return (await this._makeRequest({
-        path: "/user/repos",
+        path: `/repos/${repoFullName}/branches`,
         params: {
           per_page: 100,
           page,
         },
       })).data
     },
+    async getCommits(opts) {
+      const {
+        repoFullName,
+      } = opts
+      return (await this._makeRequest({
+        path: `/repos/${repoFullName}/commits`,
+        params: {
+          per_page: 100,
```

Comment:
no pagination support here but there is for getBranches... should be more consistent?

Reply:
fixed